build.psm1

```diff
@@ -3676,6 +3676,14 @@ function New-NugetConfigFile {
     $content += $newLine + $nugetConfigFooterTemplate
 
     Set-Content -Path (Join-Path $Destination 'nuget.config') -Value $content -Force
+
+    # Set the nuget.config file to be skipped by git
+    push-location $Destination
+    try {
+        git update-index --skip-worktree (Join-Path $Destination 'nuget.config')
```

Comment:
Can we avoid dancing with the config file?
There are some options to have some predefined nuget.config files:
1. Use `RestoreConfigFile` option in csproj/props file.  https://learn.microsoft.com/en-us/nuget/reference/msbuild-targets#restore-target
2. Use command line `dotnet restore --configfile nuget1.config` https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-restore#options

Or we could exclude the nuget.config file at all with `RestoreSources` option in csproj/props file https://learn.microsoft.com/en-us/nuget/reference/msbuild-targets#restore-target

Reply:
We have been promised a feature from the Azure Feed team that should let us move to that.  This should be considered a temporary issue.